### PR TITLE
Feat: Combobox

### PR DESCRIPTION
**Pull request by tuul-wq**
- New Combobox
- Use `@ariakit/react` to make Radix-UI Popover work with Input as trigger
- Updated `onChange` for `<Input />` to make it compatible with `@ariakit/react` and versatile overall

related to #2490 

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -67,6 +67,7 @@
   },
   "dependencies": {
     "@apollo/client": "3.11.8",
+    "@ariakit/react": "^0.4.13",
```
Comment (johnthecat) [inline package.json]: use exact version please

File: src/renderer/shared/ui-kit/Combobox/Combobox.tsx [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        </Ariakit.ComboboxProvider>
+      </RadixPopover.Root>
+    </Context.Provider>
+  );
+};
+
+type TriggerProps = {
+  placeholder?: string;
+};
+const Trigger = ({ placeholder }: TriggerProps) => {
```
Comment (johnthecat) [inline src/renderer/shared/ui-kit/Combobox/Combobox.tsx]: I think Trigger is obsolete here, just move it into `Root`

File: src/renderer/shared/ui-kit/Combobox/Combobox.tsx [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+import { cnTw } from '@/shared/lib/utils';
+import { Input } from '@/shared/ui';
+import { ScrollArea } from '../ScrollArea/ScrollArea';
+import { Surface } from '../Surface/Surface';
+import { useTheme } from '../Theme/useTheme';
+import { gridSpaceConverter } from '../_helpers/gridSpaceConverter';
+
+type ContextProps = {
+  isOpen?: boolean;
+  setIsOpen?: (open: boolean) => void;
```
Comment (johnthecat) [inline src/renderer/shared/ui-kit/Combobox/Combobox.tsx]: onToggle?

File: src/renderer/shared/ui-kit/Combobox/Combobox.tsx [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+  side?: 'top' | 'right' | 'bottom' | 'left';
```
Comment (johnthecat) [inline src/renderer/shared/ui-kit/Combobox/Combobox.tsx]: I don't think we need it, it should be calculated automatically

File: src/renderer/shared/ui-kit/Combobox/Combobox.tsx [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+  sideOffset?: number;
```
Comment (johnthecat) [inline src/renderer/shared/ui-kit/Combobox/Combobox.tsx]: Can't image a single case with this prop

File: src/renderer/shared/ui-kit/Combobox/Combobox.tsx [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+  align?: 'start' | 'center' | 'end';
```
Comment (johnthecat) [inline src/renderer/shared/ui-kit/Combobox/Combobox.tsx]: Can't image a single case with this prop

File: src/renderer/shared/ui-kit/Combobox/Combobox.tsx [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+  alignOffset?: number;
```
Comment (johnthecat) [inline src/renderer/shared/ui-kit/Combobox/Combobox.tsx]: Can't image a single case with this prop

File: src/renderer/shared/ui-kit/Combobox/Combobox.tsx [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+  testId?: string;
+  comboboxRef?: RefObject<HTMLInputElement>;
```
Comment (johnthecat) [inline src/renderer/shared/ui-kit/Combobox/Combobox.tsx:30]: Why is it a prop in `Root` component?

File: src/renderer/shared/ui-kit/Combobox/Combobox.tsx [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+  listboxRef?: RefObject<HTMLDivElement>;
```
Comment (johnthecat) [inline src/renderer/shared/ui-kit/Combobox/Combobox.tsx:31]: Same here

File: src/renderer/shared/ui-kit/Combobox/Combobox.tsx (hunk 2 above)
Comment (johnthecat) [reply src/renderer/shared/ui-kit/Combobox/Combobox.tsx]: Ok, rename it to `Input`

**Review by Asmadek — APPROVED** [review]

**Pull request merged**